astropy/table/tests/test_index.py

```diff
@@ -550,6 +551,21 @@ def test_table_index_time_warning(engine):
     assert len(wlist) == 0
 
 
+def test_index_pickle(engine):
```

Comment:
Does this test need YAML? Probably have to define `HAS_YAML` if not alreay.

```suggestion
@pytest.mark.skipif('not HAS_YAML')
def test_index_pickle(engine):
```